map/framework.cpp

```diff
@@ -1482,11 +1478,26 @@ bool Framework::Search(search::SearchParams const & params)
     intent.m_params.m_minDistanceOnMapBetweenResults = eps;
   }
 
-  intent.m_handle = m_searchEngine->Search(intent.m_params, intent.m_viewport);
+  Search(intent);
 
   return true;
 }
 
+void Framework::Search(SearchIntent & intent) const
+{
+  if (!m_isViewportInitialized)
+  {
+    Platform().RunOnGuiThread([this, &intent]
+    {
+      this->Search(intent);
```

Comment:
So, there'll be a lot of Search() tasks on the UI thread, until viewport is initialized. Please, collect search requests and post them to the search engine as soon as viewport is initialized.